File: TOMIMSV4/Entity/Mantenimientos/UnidadMedida/clsBeUnidad_medida.vb
<Serializable>
Public Class clsBeUnidad_medida
    Implements ICloneable
    Implements IDisposable

    Public Property IdUnidadMedida() As Integer = 0
    Public Property IdPropietario() As Integer = 0
    Public Property Propietario As New clsBePropietarios()
    Public Property Codigo As String = ""
    Public Property Nombre() As String = ""
    Public Property Activo() As Boolean = False
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property User_agr() As String = ""
    Public Property IsNew() As Boolean = False
    Public Property es_um_cobro As Boolean = False
    Public Property factor As Double = 1

    Sub New()
    End Sub

    Sub New(ByRef IdUnidadMedida As Integer, ByVal IdPropietario As Integer, ByVal Nombre As String, ByVal activo As Boolean, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal user_agr As String)
        Me.IdUnidadMedida = IdUnidadMedida
        Me.IdPropietario = IdPropietario
        Me.Nombre = Nombre
        Me.Activo = activo
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.User_agr = user_agr
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
        If Propietario IsNot Nothing Then
            Propietario.Dispose()
            Propietario = Nothing
        End If
    End Sub
#End Region

End Class

File: WMSWebAPI/Entity/Producto/clsBeUnidad_medida.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeUnidad_medida : ICloneable
    {
        [Column("IdUnidadMedida")]
        [DisplayName("IdUnidadMedida")]
        public int IdUnidadMedida { get; set; } = 0;

        [Column("IdPropietario")]
        [DisplayName("IdPropietario")]
        public int IdPropietario { get; set; } = 0;

        [Column("Nombre")]
        [DisplayName("Nombre")]
        public string Nombre { get; set; } = "";

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("codigo")]
        [DisplayName("codigo")]
        public string Codigo { get; set; } = "";

        [Column("es_um_cobro")]
        [DisplayName("es_um_cobro")]
        public bool Es_um_cobro { get; set; } = false;

        [Column("factor")]
        [DisplayName("factor")]
        public double Factor { get; set; } = 0;

        public clsBeUnidad_medida() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}